core/src/main/java/ai/timefold/solver/core/impl/domain/variable/cascade/CascadeUpdateElementShadowVariableDescriptor.java

```diff
@@ -0,0 +1,135 @@
+import ai.timefold.solver.core.impl.domain.common.accessor.MemberAccessorFactory;
+import ai.timefold.solver.core.impl.domain.entity.descriptor.EntityDescriptor;
+import ai.timefold.solver.core.impl.domain.policy.DescriptorPolicy;
+import ai.timefold.solver.core.impl.domain.variable.descriptor.ShadowVariableDescriptor;
+import ai.timefold.solver.core.impl.domain.variable.descriptor.VariableDescriptor;
+import ai.timefold.solver.core.impl.domain.variable.inverserelation.InverseRelationShadowVariableDescriptor;
+import ai.timefold.solver.core.impl.domain.variable.listener.VariableListenerWithSources;
+import ai.timefold.solver.core.impl.domain.variable.nextprev.NextElementShadowVariableDescriptor;
+import ai.timefold.solver.core.impl.domain.variable.nextprev.PreviousElementShadowVariableDescriptor;
+import ai.timefold.solver.core.impl.domain.variable.supply.Demand;
+import ai.timefold.solver.core.impl.domain.variable.supply.SupplyManager;
+
+public final class CascadeUpdateElementShadowVariableDescriptor<Solution_> extends ShadowVariableDescriptor<Solution_> {
+
+    private final List<VariableDescriptor<Solution_>> sourceVariableDescriptorList = new ArrayList<>();
+    private ShadowVariableDescriptor<Solution_> sourceShadowVariableDescriptor;
+    private ShadowVariableDescriptor<Solution_> nextElementShadowVariableDescriptor;
+    private MemberAccessor listenerUpdateMethod;
+
+    public CascadeUpdateElementShadowVariableDescriptor(int ordinal, EntityDescriptor<Solution_> entityDescriptor,
+            MemberAccessor variableMemberAccessor) {
+        super(ordinal, entityDescriptor, variableMemberAccessor);
+    }
+
+    @Override
+    public void processAnnotations(DescriptorPolicy descriptorPolicy) {
+        // Do nothing
+    }
+
+    @Override
+    public void linkVariableDescriptors(DescriptorPolicy descriptorPolicy) {
+        this.sourceShadowVariableDescriptor = entityDescriptor.getShadowVariableDescriptor(variableMemberAccessor.getName());
+
+        var inverseRelationShadowDescriptor = entityDescriptor.getShadowVariableDescriptors().stream()
+                .filter(variableDescriptor -> InverseRelationShadowVariableDescriptor.class
+                        .isAssignableFrom(variableDescriptor.getClass()))
+                .findFirst()
+                .orElse(null);
+        if (inverseRelationShadowDescriptor == null) {
+            throw new IllegalArgumentException(
+                    """
+                            The entityClass (%s) has an @%s annotated property (%s), but has no @InverseRelationShadowVariable shadow variable defined.
+                            Maybe add a new shadow variable @InverseRelationShadowVariable to the entity %s."""
+                            .formatted(entityDescriptor.getEntityClass(),
+                                    CascadeUpdateElementShadowVariable.class.getSimpleName(), variableMemberAccessor.getName(),
+                                    entityDescriptor.getEntityClass()));
```

Comment:
I wonder if this is necessary. If the inverse is not defined, can we not use a supply? List variable works even without defining these, so the code to support that is already in there.

Reply:
I see your point. The main reason for adding these validations is to define the [dependencies](https://github.com/TimefoldAI/timefold-solver/pull/954/files#diff-4ed0ff66350dc1ce12d0b8dd2a56f04e6410eaaeef73cc15b1a0254c08d05adfR134) that will impact the global listeners sort.

I'm not sure if it's possible to do that without the related fields.

Reply:
Don't you already know which fields this depends on?

Typically, a var listener is defined with a source variable.
In this case, the listener is _on_ the source variable, and it defines a target method.
You can think of the cascade shadow var as:

    @Cascade...(source = "var", target="run")
    Object var = ...

Therefore you should be depending on `var`, right? Or am I misunderstanding?

Reply:
I've been wondering... how can you _know_ which vars this will depend on?
You are requiring the prev/next/etc. to exist, and you're basing it on that.
But is that correct? Why not let the user decide the source?

    `@Cascade...(targetMethodName = "...", source = "...")`

Reply:
New approach implemented